buildSrc/src/main/kotlin/pklGradlePluginTest.gradle.kts

```
@@ -88,10 +88,10 @@ tasks.addRule("Pattern: compatibilityTest[All|Releases|Latest|Candidate|Nightly|
 }
 
 fun createCompatibilityTestTask(versionInfo: GradleVersionInfo): Task =
-  createCompatibilityTestTask(versionInfo.version, versionInfo.downloadUrl)
+  createCompatibilityTestTask(versionInfo.version, versionInfo.downloadUrl) as Task
```

Not sure how this cast can succeed, but, it's also not necessary; `dependsOn` can accept a `Provider<Task>`

```suggestion
  createCompatibilityTestTask(versionInfo.version, versionInfo.downloadUrl)
```